pkg/controller/deployment/deployment_checked_adapter.go

```diff
@@ -0,0 +1,41 @@
+package deployment
+
+import (
+	"context"
+	"fmt"
+
+	v1 "k8s.io/client-go/kubernetes/typed/apps/v1"
+)
+
+var globalDeploymentChecker deploymentChecker
+
+var knownCertManagerDeployments = map[string][]string{
+	//https://github.com/jetstack/cert-manager/tree/master/deploy/charts/cert-manager
+	//https://github.com/IBM/ibm-cert-manager-operator/blob/master/deploy/olm-catalog/ibm-cert-manager-operator/3.17.0/ibm-cert-manager-operator.v3.17.0.clusterserviceversion.yaml
+	"openshift-operators": {"ibm-cert-mana`ger-operator", "cert-manager"},
+	//https://github.com/komish/certmanagerdeployment-operator/blob/main/controllers/componentry/constants.go
+	"cert-manager": {"cert-manager"},
+}
+
+func InitGlobalDeploymentChecker(deploymentGetter v1.DeploymentsGetter) {
+	globalDeploymentChecker = newDeploymentChecker(deploymentGetter)
+}
```

Comment:
Let's make the deployment checker a regular library-go like controller.

Package level variables are really discouraged in golang, please see https://peter.bourgon.org/go-for-industrial-programming/ and https://peter.bourgon.org/blog/2017/06/09/theory-of-modern-go.html for more background.

Here, concretely, the package level variable is unsynchronized, nothing prevents it from being invoked multiple times, from multiple goroutines which will can cause side effects.

Reply:
i see this is using `StaticResourcesAdapterFunction` as `WithPreconfitionFunc`. This should then still be a struct, and `WithPreconditionsFunc` should reference the struct method.

Reply:
Let me look into this again, once the library-go PR gets in. 

But I'm not very keen on putting the deployment checker into the library-go. It's essentially a getter (so nothing fancy) with adapters that fit other interfaces.

Reply:
I agree on not putting deployment checker in library-go. I am just mandating against global state here.

Reply:
Got rid of the global state.